beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -140,8 +140,17 @@ func (vs *Server) GetBeaconBlock(ctx context.Context, req *ethpb.BlockRequest) (
 		// Set sync aggregate. New in Altair.
 		vs.setSyncAggregate(ctx, sBlk)
 
-		// Set execution data. New in Bellatrix.
-		if err := vs.setExecutionData(ctx, sBlk, head); err != nil {
+		// Get local and builder (if enabled) payloads. Set execution data. New in Bellatrix.
+		localPayload, err := vs.getLocalPayload(ctx, sBlk.Block(), head)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not get local payload: %v", err)
+		}
+		builderPayload, err := vs.getBuilderPayload(ctx, sBlk.Block().Slot(), sBlk.Block().ProposerIndex())
+		if err != nil {
+			builderGetPayloadMissCount.Inc()
+			log.WithError(err).Error("Could not get builder payload")
+		}
+		if err := vs.setExecutionData(ctx, sBlk, localPayload, builderPayload); err != nil {
```

Comment:
Just thinking about the future, maybe we need to implement a wrapper interface for this to get the Payload to be both payload or payloadand blobs friendly.

Reply:
I don't think we should mix Payload and Blob bundles together. They should be separated. See my implementation here: https://github.com/prysmaticlabs/prysm/pull/12454